File: Androids/Thirium 310/blue_blood_V1.py
import numpy as np
import matplotlib.pyplot as plt

# Constants
numParticles = 100  # Number of particles
timeSteps = 100  # Simulation time steps
dt = 0.01  # Time step duration

# Particle properties
mass = np.random.rand(numParticles) * 5 + 1  # Random mass between 1 and 6
charge = np.random.rand(numParticles) * 10  # Random charge
energyLevel = np.random.rand(numParticles) * 100 + 50  # Energy 50-150
position = np.random.rand(numParticles, 2) * 20 - 10  # Initial positions
velocity = np.random.rand(numParticles, 2) * 2 - 1  # Initial velocities
damagedParticles = np.zeros(numParticles)  # Track damaged particles

# Magnetic field properties (Tesla)
B = np.array([0, max(mass)])

# Neural Network Properties (Simulated Intelligence)
neuralWeights = np.random.rand(numParticles) * 5
neuralFiringThreshold = 10
neuralInput = np.zeros(numParticles)
neuralOutput = np.zeros(numParticles)

# Brownian Motion Factor (Random Motion Influence)
brownianFactor = 0.02

# Start simulation loop
plt.figure()
for t in range(timeSteps):
    # Apply forces (Electric & Magnetic Field Effects)
    Efield = charge / mass  # Electric field effect
    forceE = np.column_stack((Efield, Efield))
    forceB = np.cross(np.column_stack((velocity, np.zeros(numParticles))), [B[0], B[1], 0])
    forceB = forceB[:, :2]
    velocity += (forceE + forceB) * dt

    # Brownian motion (Random perturbations)
    velocity += (np.random.rand(numParticles, 2) - 0.5) * brownianFactor

    # Update positions
    position += velocity * dt

    # Damage simulation (Random energy loss)
    energyLoss = np.random.rand(numParticles) * 5
    energyLevel -= energyLoss
    damagedParticles[energyLevel < 30] = 1  # Mark as damaged

    # Self-healing Mechanism
    for i in range(numParticles):
        if damagedParticles[i] == 1:
            distances = np.linalg.norm(position - position[i], axis=1)
            healers = (distances < 3) & (energyLevel > 50)
            
            if np.sum(healers) > 0:
                healingEnergy = min(energyLevel[healers]) * 0.1
                energyLevel[i] += healingEnergy
                energyLevel[healers] -= healingEnergy * 0.5
                damagedParticles[i] = energyLevel[i] >= 50  # Mark as healed

    # Neural Network Processing (AI-like behavior)
    neuralInput = energyLevel / (mass + 1)
    neuralOutput = np.tanh(neuralWeights * neuralInput - neuralFiringThreshold)
    firingNeurons = neuralOutput > 0.5

    # Boost healing for active neurons (adaptive response)
    energyLevel[firingNeurons] += 5

    # Ensure energy stays within limits
    energyLevel = np.clip(energyLevel, 0, 150)

    # Real-time Visualization
    plt.clf()
    plt.scatter(position[:, 0], position[:, 1], c=energyLevel, cmap='jet', s=50, edgecolors='k')
    plt.colorbar(label='Energy Level')
    plt.title("Thirium-310 Advanced Simulation: Self-Healing, Conductivity, and Neural Response")
    plt.xlabel("X Position")
    plt.ylabel("Y Position")
    plt.pause(0.05)

plt.show()

File: Androids/Thirium 310/Blue_Blood_V1.m
%% Thirium-310 Advanced Simulation
% This script simulates Thirium-310 with electrical conductivity,
% magnetic field response, neural network properties, and self-healing.

clc; clear; close all;

%% Basic Parameters
% Number of particles (Thirium-310 units)
numParticles = 100;
timeSteps = 1000;
dt = 0.01; % Time step

% Particle properties
mass = rand(numParticles,1) * 5 + 1; % Random mass between 1 and 6
charge = rand(numParticles,1) * 10; % Random charge
energyLevel = rand(numParticles,1) * 100 + 50; % Random energy (50-150)
position = randn(numParticles,2) * 2; % Initial positions
velocity = rand(numParticles,2) * 2 - 1; % Initial velocities

damagedParticles = zeros(numParticles,1); % Track damaged particles

% Magnetic Field Strength (Tesla)
B = [0, max(mass) * 5]; % Stronger field

%% Neural Network (simulated adaptive intelligence)
neuralWeights = rand(numParticles,1) * 5;
neuralFiringThreshold = 5;
neuralInput = zeros(numParticles,1);
neuralOutput = zeros(numParticles,1);

figure;
for t = 1:timeSteps
    % Apply forces (self-repairing and interaction behavior)
    Efield = charge ./ mass; % Electric field effect
    forceE = Efield .* [1,1];
    % Convert velocity into 3D space by appending a zero z-component
    velocity3D = [velocity, zeros(numParticles,1)];
    
    % Convert B into a 3D vector
    B3D = repmat([B(1), B(2), 0], numParticles, 1);
    
    % Compute cross product
    forceB3D = cross(velocity3D, B3D);
    
    % Extract only the x-y components
    forceB = forceB3D(:,1:2);
    velocity = velocity + (forceE + forceB) * dt;
    
    % Update positions
    position = position + velocity * dt;
    
    % Damage simulation (random energy loss)
    energyLoss = energyLevel * 0.005; % Only 0.5% lost per step
    energyLevel = energyLevel - energyLoss;
    damagedParticles(energyLevel < 30) = 1; % Mark particles as damaged

    energyLevel(energyLevel < 5) = 5; % Prevent particles from reaching zero energy
    
    % Self-healing Mechanism
    for i = 1:numParticles
        if damagedParticles(i) == 1
            distances = sqrt(sum((position - position(i,:)).^2,2));
            healers = (distances < 5) & (energyLevel > 30); % More particles qualify
    
            if sum(healers) > 0
                healingEnergy = mean(energyLevel(healers)) * 0.3; % Boost healing efficiency
                energyLevel(i) = energyLevel(i) + healingEnergy;
                energyLevel(healers) = energyLevel(healers) - healingEnergy * 0.1; % Reduce drain on healers
          
                % Reduce the amount of energy given per step
                healingEnergy = healingEnergy * 0.05; % Healing slowed down to 5% per step
                
                % Reduce energy loss per step
                energyLoss = energyLoss * 0.1; % Slower decay
                
                % Cap the maximum energy level per particle
                energyLevel = min(energyLevel, 100);  % Max limit of 100 (adjust as needed)

                damagedParticles(i) = energyLevel(i) >= 50; % Mark as healed
                energyLevel = min(energyLevel + rand(numParticles,1) * 2, 150); % Small boost per step
            end
        end
    end
    
    % Neural Network Processing (AI-like behavior)
    neuralInput = energyLevel ./ (mass + 1);
    neuralOutput = tanh(neuralWeights .* neuralInput - neuralFiringThreshold);
    firingNeurons = neuralOutput > 0.5;

    % Boost healing for active neurons (adaptive response)
    energyLevel(firingNeurons) = energyLevel(firingNeurons) + 5;
    
    energyLevel(isnan(energyLevel) | isinf(energyLevel)) = 50; % Reset invalid values

    % Plot real-time visualization
    scatter(position(:,1), position(:,2), 50, energyLevel, 'filled');
    colormap(jet);
    colorbar;
    title('Thirium-310 Simulation: Self-Healing, Conductivity, and Neural Response');
    xlabel('X Position'); ylabel('Y Position');
    drawnow;
end
